x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/download_source_flyout/use_delete_download_source.tsx

```diff
@@ -25,53 +25,65 @@ const ConfirmTitle = () => (
 
 interface ConfirmDeleteDescriptionProps {
   downloadSource: DownloadSource;
-  agentCount: number;
-  agentPolicyCount: number;
+  agentCount?: number;
+  agentPolicyCount?: number;
 }
 
 const ConfirmDeleteDescription: React.FunctionComponent<ConfirmDeleteDescriptionProps> = ({
   downloadSource,
   agentCount,
   agentPolicyCount,
-}) => (
-  <FormattedMessage
-    id="xpack.fleet.settings.deleteDowloadSource.confirmModalText"
-    defaultMessage="This action will delete {downloadSourceName} agent binary source. It will update {policies} and {agents}. This action can not be undone. Are you sure you wish to continue?"
-    values={{
-      downloadSourceName: <strong>{downloadSource.name}</strong>,
-      agents: (
-        <strong>
-          <FormattedMessage
-            id="xpack.fleet.settings.deleteDowloadSource.agentsCount"
-            defaultMessage="{agentCount, plural, one {# agent} other {# agents}}"
-            values={{
-              agentCount,
-            }}
-          />
-        </strong>
-      ),
-      policies: (
-        <strong>
-          <FormattedMessage
-            id="xpack.fleet.settings.deleteDowloadSource.agentPolicyCount"
-            defaultMessage="{agentPolicyCount, plural, one {# agent policy} other {# agent policies}}"
-            values={{
-              agentPolicyCount,
-            }}
-          />
-        </strong>
-      ),
-    }}
-  />
-);
+}) =>
+  agentCount !== undefined && agentPolicyCount !== undefined ? (
+    <FormattedMessage
+      id="xpack.fleet.settings.deleteDowloadSource.confirmModalText"
+      defaultMessage="This action will delete {downloadSourceName} agent binary source. It will update {policies} and {agents}. This action can not be undone. Are you sure you wish to continue?"
+      values={{
+        downloadSourceName: <strong>{downloadSource.name}</strong>,
+        agents: (
+          <strong>
+            <FormattedMessage
+              id="xpack.fleet.settings.deleteDowloadSource.agentsCount"
+              defaultMessage="{agentCount, plural, one {# agent} other {# agents}}"
+              values={{
+                agentCount,
+              }}
+            />
+          </strong>
+        ),
+        policies: (
+          <strong>
+            <FormattedMessage
+              id="xpack.fleet.settings.deleteDowloadSource.agentPolicyCount"
+              defaultMessage="{agentPolicyCount, plural, one {# agent policy} other {# agent policies}}"
+              values={{
+                agentPolicyCount,
+              }}
+            />
+          </strong>
+        ),
+      }}
+    />
+  ) : (
+    <FormattedMessage
+      id="xpack.fleet.settings.deleteDowloadSource.confirmModalTextWithoutCount"
+      defaultMessage="This action will delete {downloadSourceName} agent binary source. It will update related policies and agents. This action can not be undone. Are you sure you wish to continue?"
```

Comment:
I'd suggest:
`This action will delete {downloadSourceName} agent binary source and it will update its related policies and agents. This action can not be undone. Are you sure you wish to continue?`